src/backend/process/Heap.js

```diff
@@ -0,0 +1,15 @@
+define(function() {
+
+    function Heap(memory) {
+        this.memory = memory;
+    }
+
+    Heap.prototype.alloc = function alloc() {
+
```

Comment:
Throw an "not implemented" error instead of leaving code that should do something but does nothing.